src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -76,9 +80,9 @@ public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
 
     @Override
     public void setInstallReferrer(Intent intent) {
-        if (intent.getData() != null && intent.getData().getQueryParameter("btn_ref") != null) {
-            logDebug("Tracking incoming intent as %s", intent.toString());
-            ButtonMerchant.trackIncomingIntent(getContext(), intent);
+        if (intent.getExtras() != null && intent.hasExtra("referrer")) {
```

Comment:
Is this line needed? The merchant library should handle all of this

Reply:
The library [does not](https://github.com/button/button-merchant-android/blob/master/button-merchant/src/main/java/com/usebutton/merchant/ButtonInternalImpl.java#L79) have a check for this. Is there a scenario where the intent would be valid without a referrer?

Reply:
Yeah this line should be removed completely and there should only be a null check on the intent